blockchain/blockchain.go

```diff
@@ -729,29 +697,56 @@ func (bc *blockchain) ExecuteContractRead(ex *action.Execution) ([]byte, error)
 //======================================
 // private functions
 //=====================================
+
+func (bc *blockchain) validateBlock(blk *Block) error {
+	if bc.validator == nil {
+		logger.Panic().Msg("no block validator")
+	}
+
+	// replacement logic, used to replace a fake old dummy block
```

Comment:
we don't have dummy block now right? why we still need this?

Reply:
We still have dummy block logic in the code, but just disabled in production